### PR TITLE
BaSP-TG-116: TimeSheet form update

### PR DESCRIPTION
**Pull request by fede230298**
REPLACE form logic to update it and use React hook form
ADD Joi validations to check inputs

### CODE REVIEW

File: src/Components/TimeSheets/Form/index.js [hunk 1]
```diff
@@ -7,20 +7,16 @@ import DatePicker from 'Components/Shared/Datepicker';
...
 import { useSelector, useDispatch } from 'react-redux';
+import { getOneTimeSheet, postTimeSheet, putTimeSheet } from '../../../redux/timeSheets/thunks';
+import { getEmployees } from '../../../redux/employees/thunks';
+import { getProjects } from '../../../redux/projects/thunks';
+import { getTasks } from '../../../redux/tasks/thunks';
 import { useForm } from 'react-hook-form';
-import { getOneTimeSheet, postTimeSheet, putTimeSheet } from 'redux/timeSheets/thunks';
-import { getEmployees } from 'redux/employees/thunks';
-import { getProjects } from 'redux/projects/thunks';
-import { getTasks } from 'redux/tasks/thunks';
```
Comment (JuliMonte5) [inline src/Components/TimeSheets/Form/index.js:14]: Use absoulte routes here

Comment (fede230298) [reply src/Components/TimeSheets/Form/index.js]: Done!

**Review by GonzaloAsencio — APPROVED** [review]

**Review by JuliMonte5 — APPROVED** [review]

File: src/Components/TimeSheets/Form/index.js [hunk 2]
```diff
@@ -35,7 +31,24 @@ const TimeSheetsForm = () => {
...
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset
+  } = useForm({
+    mode: 'onChange',
+    resolver: joiResolver(timeSheetValidation)
+  });
+
+  const MOCK_DATA = {
```
Comment (Ignacio-Sormani) [inline src/Components/TimeSheets/Form/index.js:44]: Change this name, this is not a data mocked. It's real data. Also, it should be in camelCase

Comment (fede230298) [reply src/Components/TimeSheets/Form/index.js]: Done!

**Review by sofihoj — APPROVED** [review]

**Pull request merged**